language-design.md

```diff
@@ -17,6 +17,97 @@ For each quote, describe what it was about the quote that led you pick it.
+time. This is the result of how people begin to write and speak in different ways depending on 
+their cultural context, and sometimes, different dialects and possibly even new languages can emerge. 
+In the same vein, programming languages are not confined to the rules that were established by 
+their initial founders - in fact, a programming language may evolve so much over time so that it 
+could no longer be credited to the few individuals that ‘created’ it. Rather, a programming language 
+becomes the creation of the entire community of programmers that have written in the language, 
+created useful tools with it, and shared their experiences and solutions with others. This is 
+more in line with the ideas that are suggested by Steele in his talk, Growing a Language.
+
+However, in the context of the article, this quote suggests that these communities also have 
+the danger of being exclusive and producing a bias that a certain programming language might 
+only be used by certain people for certain functions, and in that sense, perhaps programming 
+languages may not be as democratic as they may seem. Particularly, as a women in the field of 
+Computer Science, I feel intimidated by certain languages and I do feel like that stems from 
+the preconceived notions that only geniuses (read: genius males) can program in more “difficult” 
+and obscure languages. Yang and Rabkin suggests, then, the importance of removing “hidden biases” 
+that prevent some languages from being as accessible as they can be, and this is something that 
+I wholeheartedly agree with. 				
+
+--
+	
+*“To add all these types [interval, vector, matrix] to the Java programming language would be 
+just too much. Some parts of the programming vocabulary are fit for all programmers to use, but 
+other parts are just for their own niches. It would not be fair to weigh down all programmers with 
+the need to have or to learn all the words for all niche uses. We should not make the Java programming 
+language a cathedral, but a plain bazaar might be too loose. What we need is more like a shopping 
+mall, where there are not quite as many choices but most of the goods are well designed and sellers 
+stand up and back what they sell.” [Steele, 1998]*
+
+I do not understand this phrase: “It would not be fair to weigh down all programmers with the need 
+to have or to learn all the words for all niche uses.” No one is requiring a user to understand all 
+the words for niche uses before the user can use the language for his/her own purposes. The words 
+for niche uses can exist in the language, accessible for all to use, and the people who don’t know 
+what they are will never use them anyway. At the same time, it is understandable that there needs 
+to be a some degree of limited expressiveness with the language, but flexibility and giving the users
+the option to use certain features is important as well. There does need to be a balance between the two, 
```

Comment:
Do you think that a good way to achieve this balance is by using libraries / packages?  Perhaps by taking many of the words that are only useful in niche cases and bundling related terms into a package or library, then most individuals will not be bothered by these terms (via name clashes or otherwise) but any individual who still needs the use of the words can import the package for their own niche and learn all the words they need to for their own niche?  Or perhaps there is a better way of accomplishing this goal?